### DISCUSSION TITLE
What do you think about giving the possibility to disable all collaborative features and hide the Sign In button

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#16822] [Show and tell]
Issue: #12325 

It would be nice to have a property in the settings file that could hide this element.

![image](https://github.com/zed-industries/zed/assets/20476002/0ad1772e-69c5-4580-8537-41e55ccb55b8)

---

Issue: #13218

 I'd like to see an option added to disable all collaborative features of Zed. For users like me who don’t need these, they take up unnecessary space (and resources?). While hiding the buttons for the collaboration, chat, and notification panels through settings is possible, I don't think it's enough.

• Completely disable collaboration and chat panels: Even when hidden, these panels might still be running in the background, using resources?
• Suppress collaboration-related notifications: Invites, calls, mentions, and other notifications can be distracting when collaboration is not in use. Currently, they still generate system activity and interruptions.
• Repurpose the top bar: The top bar currently shows user avatars and speaker highlights, which are redundant for solo users. This space could be better utilized for other purposes.

Hiding buttons is a step in the right direction, but fully disabling these features would create a cleaner, more efficient workspace for those of us who use Zed as a standalone editor. It would improve the user experience by providing a distraction-free environment and optimizing resource usage.

----

It would be appreciated to post a reaction and vote if you agree with the possibility of disabling the collaborative function.

**Comment by mrhillsman** [comment]
Clicked on Collab Panel button and was not expecting to be automatically shown a list of folks who I don't know and what seems to be a connection made to where I don't know either. Unfortunately this implementation of collaboration seems a bit too suspect even though I was intrigued to use this editor daily this seems like something waiting to be exploited or backdoored. Would be nice if the collaboration was optional and did something like offered the ability to send a private link for collaboration or a sign-in/out option so as a user I know I don't have background connections to services I am unaware of. Unfortunately uninstalling until a better implementation is done on this front.

**Comment by defulmere** [comment]
From #13218 

> In the similar vein as https://github.com/zed-industries/zed/issues/12325, it's rather unlikely that we'll ever support this.
> 
> At its core Zed is a collaborative code editor, so we don't want to hide the collaborative features.

It's a shame the devs have taken this stance. As a user I find that, at its core, Zed is a **very** fine editor while the collaboration features are fluff that I'll never use. I'd love to be able to use Zed without a persistent sign-in button cluttering up the UI.

**Reply by injust** [reply]
@defulmere https://github.com/zed-industries/zed/pull/30450

**Comment by Askath** [comment]
Would also love to have the option as our company prevents web socket connections and thus collaboration features don't work anyway and just spam the log. 

So being able to disable those would remove some annoyance for me

**Comment by Twig6943** [comment]
Any reason not to do with since it got 100+ upvotes

**Reply by nikdoof** [reply]
Zed see it as a core feature they're not going to disable. Much like AI support which takes up more of the front page of the website than their collab features, which happily has a `disable_ai` feature switch. 🤷‍♂️

**Comment by Narann** [comment]
I use Zed as wonderful text editor. I don't need to "Sign In" somewhere.
I would love to remove anything related to Collaboration, but if not possible, I would simply like to hide it.
Thanks,
